mopidy_alsamixer/mixer.py

```diff
@@ -173,36 +215,43 @@ def trigger_events_for_changed_values(self):
             self.trigger_mute_changed(self._last_mute)
 
 
-class AlsaMixerObserver(threading.Thread):
-    daemon = True
-    name = "AlsaMixerObserver"
+class AlsaMixerObserver(PollingActor):
 
-    def __init__(self, device, control, callback=None):
-        super().__init__()
-        self.running = True
+    name = "alsamixer-observer"
 
-        # Keep the mixer instance alive for the descriptors to work
-        self.mixer = alsaaudio.Mixer(device=device, control=control)
+    combine_events = True
 
-        descriptors = self.mixer.polldescriptors()
-        assert len(descriptors) == 1
-        self.fd = descriptors[0][0]
-        self.event_mask = descriptors[0][1]
+    def __init__(self, mixer, parent):
+        # Note: ALSA mixer instance must be kept alive
+        # to keep poll descriptors open
+        self._mixer = mixer
+        self._parent = parent
 
-        self.callback = callback
+        # FIXME: Remove when a new version of pyalsaaudio is released
+        # See https://github.com/larsimmisch/pyalsaaudio/pull/108
+        def check_fd(fd):
+            return fd != -1 and fd != struct.unpack("I", b"\xFF\xFF\xFF\xFF")[0]
 
-    def stop(self):
-        self.running = False
+        super().__init__(
+            fds=tuple(
+                (fd, event_mask | select.EPOLLET)
+                for (fd, event_mask) in self._mixer.polldescriptors()
+                if check_fd(fd)
+            )
+        )
```

Comment:
It isn't super clear exactly what code needs updating (and how) when a new version of pyalsaaudio is released.

Is this code forwards-compatible, or will it break with newer versions of pyalsaaudio?

Reply:
Added clarifications. It's compatible with newer versions, I tried to underline it now phrasing it as "it can be removed".

Compatibility follows from the fact that max absolute value of correct file descriptor data type -- signed int32 -- is about a half of unsigned int32 (used in current latest release). So this check essentially becomes just garbage.